### PR TITLE
Skip optional plugins while running Kedro Viz

### LINKED ISSUE #1159: Kedro-viz needs too much to run properly
## Description

So according to this https://github.com/kedro-org/kedro-viz/issues/1125 kedro-viz requires 
- all plugins to be working (incl. mlflow connection to remote tracking server if leveraged)
- all python packages to be installed

For me it just doesn't make sense. Why would I need to have the python env ready to visualize project flow? The original issue seems dead, but kedro-viz is still in some way a core kedro feature, so..

### PR DESCRIPTION
**Pull request by ravi-kumar-pilla**
## Description

Resolves #1159 

## Development notes

* Add additional option during kedro viz run to ignore plugins
* Modify kedro-dataset-stats hook name
* Refactor public method names to private

## QA notes
1. Install local kedro Viz package by executing `pip install -e package`
2. Navigate to demo project `cd demo-project`
3. Install kedro mlflow `pip install kedro-mlflow`
4. Run kedro viz without ignoring plugins - `kedro viz`
5. You will find logs in the terminal related to mlflow plugin
6. Re-run kedro viz with --ignore-plugins flag - `kedro viz --ignore-plugins`
7. There should not be any logs related to mlflow.

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [x] Added tests to cover my changes

<a href="https://gitpod.io/#https://github.com/kedro-org/kedro-viz/pull/1544"></a>

### CODE REVIEW

**Comment by ravi-kumar-pilla** [comment]
Hi @tynandebold , @rashidakanchwala, @noklam 

This PR is still in draft phase (I need to add pytests for the same). Before I do that, I would like to get some feedback with the approach here. This is from Nok's suggestion [here](https://github.com/kedro-org/kedro-viz/issues/1159#issuecomment-1310340775)

To summarize, I have added an additional option of `--ignore-plugins=True` while running Kedro viz via cli. This unregisters all plugins other than the kedro viz mandatory plugins which is `kedro-telemetry` for now. Please let me know your thoughts or suggestions. Thank you

**Comment by noklam** [comment]
Comments:
- `kedro-telemetry` is not a mandatory requirements, it should not fail if such plugin doesn't exist
- Is there any value to run plugins at all, or should we default `--ignore-plugin yes`?
- I actually think the original proposal with `_NullManager()` is sufficient, it should still track `kedro viz` telemetry. You can verify this with HEAP, if you are not familiar with it maybe ask @tynandebold how to test it with the development environment. I vaguely remember there are guides in confluence too if you search "HEAP" in "kedro" confluence.

**Comment by ravi-kumar-pilla** [comment]
> Comments:

> * `kedro-telemetry` is not a mandatory requirements, it should not fail if such plugin doesn't exist

Yes it is true. But, I felt we should have kedro telemetry to collect the data during a kedro viz run. I will confirm this with @tynandebold 

> * Is there any value to run plugins at all, or should we default `--ignore-plugin yes`?

In future when vizro integration is complete, we will have it as plugin and there should be some way to not ignore few mandatory plugins.

> * I actually think the original proposal with `_NullManager()` is sufficient, it should still track `kedro viz` telemetry. You can verify this with HEAP, if you are not familiar with it maybe ask @tynandebold how to test it with the development environment. I vaguely remember there are guides in confluence too if you search "HEAP" in "kedro" confluence.

I need to set this up locally. I will get in touch with Tynan or if someone points me to the doc for setup. Thank you

**Comment by ravi-kumar-pilla** [comment]
Hi @noklam ,

What you said is right regarding kedro telemetry. I can see network requests even after unregistering the hook. I think the data is collected from the kedro side and viz actually does not need the plugin. I have modified the code to initialize `_NullPluginManager()` as suggested. 

But I am keeping the --ignore-plugins flag default to `False` as I see there are some meta files being generated when executing kedro viz with mlflow installed. Also this will help when we need vizro plugin to be mandatory. Thank you

File: package/setup.py [hunk 1]
```diff
@@ -47,7 +47,7 @@
         "kedro.global_commands": ["kedro-viz = kedro_viz.launchers.cli:commands"],
         "kedro.line_magic": ["line_magic = kedro_viz.launchers.jupyter:run_viz"],
         "kedro.hooks": [
-            "plugin_name = kedro_viz.integrations.kedro.hooks:dataset_stats_hook"
+            "kedro-dataset-stats = kedro_viz.integrations.kedro.hooks:dataset_stats_hook"
```
[the remark this answers is not in the document]
Comment (ravi-kumar-pilla) [reply package/setup.py:50]: okay, I followed the plugin names we already had for example (kedro-telemetry, kedro-mlflow) to be consistent

**Review by rashidakanchwala — APPROVED** [review]
LGTM. thanks

**Review by tynandebold — APPROVED** [review]
This looks good to me.

In the future, we could add something to the docs, akin to [this](https://docs.kedro.org/en/stable/visualisation/kedro-viz_visualisation.html#automatic-visualisation-updates). We don't do that with every CLI option we have though, so for now, this is good to go.

File: package/kedro_viz/integrations/kedro/data_loader.py [hunk 2]
```diff
@@ -32,10 +33,26 @@
...
+    """
+
+    def __init__(self, *args, **kwargs):
+        pass
+
+    def __getattr__(self, name):
+        return self
+
+    def __call__(self, *args, **kwargs):
+        pass
```
Comment (noklam) [inline package/kedro_viz/integrations/kedro/data_loader.py:55]: Is this needed? What's the different between this and the `NullPluginManager`.

**Comment by tynandebold** [comment]
Also, please don't forget to add a line to the release notes.

File: package/kedro_viz/integrations/kedro/data_loader.py (hunk 2 above)
Comment (ravi-kumar-pilla) [reply package/kedro_viz/integrations/kedro/data_loader.py:55]: No difference really but - _NullPluginManager() was not introduced in kedro 0.17.5 which is the oldest kedro version that viz supports. So I had to create this.

**Pull request merged**